x/rollapp/keeper/msg_server_update_state_test.go

```diff
@@ -147,55 +146,31 @@ func (suite *RollappTestSuite) TestUpdateState() {
 		suite.Require().Nil(err)
 
 		// end block
-		responseEndBlock := suite.App.EndBlocker(suite.Ctx, abci.RequestEndBlock{Height: suite.Ctx.BlockHeight()})
+		suite.App.EndBlocker(suite.Ctx, abci.RequestEndBlock{Height: suite.Ctx.BlockHeight()})
+
+		for i := uint64(0); i <= expectedLatestStateInfoIndex.Index; i++ {
+			expectedStateInfo, found := suite.App.RollappKeeper.GetStateInfo(suite.Ctx, rollapp.GetRollappId(), i)
+			if found {
+				if uint64(suite.Ctx.BlockHeight()) > disputePeriodInBlocks {
```

Comment:
the blokHeight AFAIU shouldn't change between loop iteratins. why is not checked before the loop?

Reply:
true. moved outside the loop